### PR TITLE
Start Blobstore functionality

### PR DESCRIPTION
**Pull request by lbourret**
-get blobstore URL
-get file URL
-no access to file URL yet

### CODE REVIEW

**Comment by lbourret** [comment]
in process. still need to access file URL  & put file URL in comment object

File: portfolio/src/main/webapp/contact.html [hunk 1]
```diff
@@ -49,6 +49,11 @@ <h2>COMMENT BELOW</h2>
                     <p>COMMENT</p>
                     <input name="text-input" required>
                     <br/><br/>
+                    <p>IMAGE UPLOAD</p>
+                    <input type="file" name="image">
+                    <button id="fileButton" onclick="getFileURL();return false;">UPLOAD</button>
```
Comment (meltsufin) [inline portfolio/src/main/webapp/contact.html:54]: Note sure what `getFileURL()`, but the name doesn't sound right here. Maybe, `uploadFile()`?

File: portfolio/src/main/java/com/google/sps/servlets/BlobstoreServlet.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ */
+@WebServlet("/blobstore-upload-url")
+public class BlobstoreServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    String uploadUrl = blobstoreService.createUploadUrl("/form-handler");
+
+    System.out.println("url:" + uploadUrl);
```
Comment (elefeint) [inline portfolio/src/main/java/com/google/sps/servlets/BlobstoreServlet.java:38]: Is the logging left in intentionally?

File: portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+public class FormHandlerServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get("myFile");
+
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      response.getWriter().println("Image Error");
```
Comment (elefeint) [inline portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java:51]: What if the user intentionally did not upload a file? Is such a scenario possible?

**Comment by lbourret** [comment]
Small style issue- I'm not sure how to style only the file upload button and not the entire element

**Comment by meltsufin** [comment]
> Small style issue- I'm not sure how to style only the file upload button and not the entire element

Googling around I found [this trick](https://stackoverflow.com/questions/572768/styling-an-input-type-file-button/25825731#25825731).

File: portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java [hunk 4]
```diff
@@ -85,4 +92,30 @@ private String getParameter(HttpServletRequest request, String name, String defa
     }
     return value;
   }
+
+    /**
+     *
+     * @return url with blob key
+     */
+    private String getBlobKey(HttpServletRequest request) {
```
Comment (meltsufin) [inline portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java:100]: Spacing off?

File: portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java [hunk 5]
```diff
@@ -55,28 +60,30 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       return;
     }
 
-    // Get the input from form text box
+    // Get the input from comment form 
     String text = getParameter(request, "text-input", null);
     String username = getParameter(request, "username", null);
     String email = userService.getCurrentUser().getEmail();
     long timestamp = System.currentTimeMillis();
+    String image = getBlobKey(request);
```
Comment (meltsufin) [inline portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java:68]: I like that you put this functionality of extracting the URL for the image in a separate method.

File: portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java [hunk 6]
```diff
@@ -45,12 +45,26 @@
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
     Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
-    List<BlobKey> blobKeys = blobs.get("myFile");
+    List<BlobKey> blobKeys = blobs.get("image");
 
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
     if (blobKeys == null || blobKeys.isEmpty()) {
-      response.getWriter().println("Image Error");
+        System.out.println("hi");
```
Comment (elefeint) [inline portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java:52]: Remove debug messages?

**Pull request merged**